detox/src/Detox.js

```diff
@@ -248,10 +248,17 @@ class Detox {
       .map(0)
       .value();
 
+    const optimizeAppInstall = this._behaviorConfig.optimizeAppInstall;
+
     for (const appName of appNames) {
       await this.device.selectApp(appName);
-      await this.device.uninstallApp();
-      await this.device.installApp();
+
+      if (optimizeAppInstall) {
+        await this.device.resetAppState();
+      } else {
+        await this.device.uninstallApp();
+        await this.device.installApp();
+      }
```

Comment:
I think it's wrong to perform this check in multiple places ([example](https://github.com/wix/Detox/pull/3195/files#diff-61fe553dde3e9ac2a314b747cf90bcadbdc0466427bced4f164bf4a924042664R322)).

If I understand correctly, the very point behind `resetAppState` was to either clear data if the `optimizeAppInstall` flag is set, or reinstall otherwise. Therefore, this code should be inside `resetAppState()` itself, shouldn't it?

Reply:
Good catch, but now I have an opposite question. Why does `resetAppState` install the app as a side effect at all? 🤔

Reply:
I think this place has a logic error. Something is wrong with handling `--reuse` flag and `--optimize-app-install` together, it is itching in the back of my head. 🤔